music/src/music_rnn/model.ts

```diff
@@ -317,21 +317,20 @@ export class MusicRNN {
       if (i < length) {
         nextInput = splitInputs[i];
       } else {
-        const logits =
-            lastOutput.matMul(this.lstmFcW).add(this.lstmFcB) as tf.Tensor2D;
-
-        let sampledOutput;
-        if (returnProbs || temperature) {
-          const theseProbs = temperature ?
-              tf.softmax(logits.div(tf.scalar(temperature))) :
-              tf.softmax(logits);
-          probs.push(theseProbs as tf.Tensor1D);
-          sampledOutput =
-              tf.multinomial(theseProbs as tf.Tensor2D, 1, undefined, true)
-                  .as1D();
+        let logits = lastOutput.matMul(this.lstmFcW).add(this.lstmFcB).as1D();
+
+        let sampledOutput: tf.Tensor1D;
+        if (temperature) {
+          logits = logits.div(tf.scalar(temperature));
+          sampledOutput = tf.multinomial(logits, 1).as1D();
         } else {
-          sampledOutput = logits.argMax(1).as1D();
+          sampledOutput = logits.argMax().as1D();
```

Comment:
wait, we don't want softmax here anymore?

Reply:
softmax doesn't change the relative magnitudes so we don't need it before an argmax